crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1521,6 +1535,9 @@ impl<'db> TypeInferenceBuilder<'db> {
     }
 
     fn module_ty_from_name(&self, module_name: ModuleName) -> Option<Type<'db>> {
+        // REVIEW: There was an opportunity here to replace `Option<Type>` with `Type` by returning
+        // `Type::Unknown(UnknownTypeKind::UnresolvedImport)` instead of `None`.
+        // I'm not including this to simplify the PR, could be done later if deemed useful.
```

Comment:
Yeah, maybe we'll want to do this, but I'm not convinced we should carry semantic meaning through the Unknown kind this way, I'd rather keep it just for debugging and perhaps improved diagnostics.